manifests/importer.pp

```diff
@@ -20,16 +20,43 @@
   ################### 3. Telemetry dashboard ########################################
   # If using puppet_metrics_dashboard:
 
-    #   class { 'puppet_metrics_dashboard':
-      # add_dashboard_examples => true,
-      # overwrite_dashboards   => false,
-      # configure_telegraf     => true,
-      #enable_telegraf        => true,
-      #master_list            => $infranode,
-      #puppetdb_list          => [$pdb],
-      #postgres_host_list     => [$postgres],
-
-      #}
+  #
+  # this is where lists of master_list , puppetdb_list, postgres_host_list hosts are found using a puppetdb query
+  #
+
+  function puppet_enterprise::active_puppetdb_hosts() {
+    if $settings::storeconfigs {
+      $pdb =
+                  puppetdb_query('resources[certname] {
+                      type = "Class" and
+                      title = "Puppet_enterprise::Profile::Puppetdb" and
+                      nodes {
+                        deactivated is null and
+                        expired is null
+                      }
+                    }').map |$data| { $data['certname'] }
+    } else {
+      $pdb = []
+  }
+
+  pe_sort($pdb)
+}
+
+
+  #make sure each list is same format
+
+  #conditions of whether postgres is present or not
+
+  #in the below class 
+    class { 'puppet_metrics_dashboard':
+      add_dashboard_examples => true,
+      overwrite_dashboards   => false,
+      configure_telegraf     => true,
+      enable_telegraf        => true,
+      master_list            => $infranode,
+      puppetdb_list          => [$pdb],
+      postgres_host_list     => [$postgres],
```

Comment:
None of these variables (`$infranode`, `$pdb` or `$postgres`) have actually been defined anywhere as far as I can tell so i think they will all be `undef`. @liamjohnsexton has this been tested?

Reply:
No, not yet - I'm actually a little bit unclear on the best way to test all this :)

Reply:
Yeah I reckon this might be something that all of the devs should sit down and work out, then document. No need for everyone to work out their own way of testing it. @MartyEwings what do you think?

Reply:
yeah we need to, testing is an issue, in theory as long as you have the metrics collector module installed any code you write that configures that can be tested, stand alone

```
      master_list            => $infranode,
      puppetdb_list          => [$pdb],
      postgres_host_list     => [$postgres],
```

those parameters for attributes where taken from my old code, to use them here, you would need to be defining them ideally through the output of a function